### PR TITLE
Feat(eos_cli_config_gen): Add router path-selection feature

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

The PR is ready but the reviewers should react upon schema definition.

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

```
    router_path_selection:
      peer_dynamic_source: <str>
      path_groups:
        - name: <str>
          id: <int>
          ipsec_profile: <str>
          flow_assignment: <str>
          local_interfaces:
            - name: <str>
              public_address: <str>
              stun:
                server_profile: <str>
          local_ips:
            - ip_address: <str>
              public_address: <str>
          dynamic_peers:
            ip_local: <bool>
            ipsec_disabled: <bool>
          static_peers:
            - router_ip: <str>
              name: <str>
              ip_address: <str>
      load_balance_policies:
        - name: <str>
          path_groups:
            - <str>
      policies:
        - name: <str>
          default_match:
            load_balance: <str>
      vrfs:
        - name: <str>
          path_selection_policy: <str>
```

## How to test

molecule

### User Checklist

- [x] - need to agree on path-group `peer dynamic` and `peer static` CLI - for now using `dynamic_peers` dict and `static_peers` list
- [x] Need clarifty on local interfaces (Ethernet 1-6 but description says subinterfaces) and `Switch` auto-completion needs some more understanding. 
    * Checked on Cloud EOS running 4.30.1F => **not implementing Switch**:
          
          test-site101-r1(config-path-group-gmuloc-test-0)#local ?
          interface  interface
             ip         ip
          
          test-site101-r1(config-path-group-gmuloc-test-0)#local interface ?
          Ethernet  Ethernet interface
          
          test-site101-r1(config-path-group-gmuloc-test-0)#local interface Ethernet ?
          <1-3>  Ethernet Port number
          

- [x] `id` for `path-group` is optional it seems so need to remove secondary key in schema maybe

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2 [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+{%             endif %}
+{%         endfor %}
+{#      peer dynamic #}
+{%         if path_group.dynamic_peers is arista.avd.defined %}
+      !
+      peer dynamic
+{%             if path_group.dynamic_peers.ip_local is arista.avd.defined(true) %}
+         ip local
+{%             endif %}
+{%             if path_group.dynamic_peers.ipsec_disabled is arista.avd.defined(true) %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2]: this should be updated to this:
- if set to true set `ipsec`
- if set to false set `ipsec disable`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                        description: |
+                          STUN server-profile names
+                        items:
+                          type: str
+            dynamic_peers:
+              type: dict
+              description: |
+                - Flow assignement LAN can not be configured in a path group with dynamic peer
+              keys:
+                ip_local:
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml:95]: add:

```
enabled:
  type: bool
```

**Comment by carlbuchmann** [comment]
Regarding this task: 

```
need to agree on path-group peer dynamic and peer static CLI - for now using dynamic_peers dict and static_peers list
```
For context here is the EOS CLI:
```
router path-selection
   peer dynamic source stun
   !
   path-group PG-1 id 666
      !
      peer dynamic
         ip local
         ipsec disabled
      !
      peer static 172.16.1.42
         ipv4 address 192.168.1.42
      !
      peer static 172.16.42.42
         name TEST-STATIC-PEER-WITH-NAME
         ipv4 address 192.168.42.4
```

I agree on naming these keys `dynamic_peers` and `static_peers`. It doesn't follow exacly the cli, but I beleive in this case would actually make more common sense for user leveraging the models direcly. we need to define multiple static peers, so it makes sense to reverse the naming compared to the EOS CLI.


We can capture the EOS specific cli in the description example:

```yaml
dynamic_peers:
  keys:
    enabled:
      type: bool
      description: Enable `peer dynamic`
```

@ClausHolbechArista what are your thoughs?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2 (hunk 1 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2]: [a8bcad1](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/a8bcad1df5579a690fc2a3dc5780901f6e9f8763)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml (hunk 2 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml:95]: [a8bcad1](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/a8bcad1df5579a690fc2a3dc5780901f6e9f8763)

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/router-path-selection.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+   ip address 10.73.255.122/24
+```
+
+### Router Path-selection
+
+#### Router Path-selection Summary
+
+| Setting | Value |
+| ------  | ----- |
+| Dynamic peers source | stun |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/router-path-selection.md]: STUN, LAN and other abbreviations should be upper case.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  router_path_selection:
+    type: dict
+    description: Dynamic path selection configuration
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
    description: Dynamic path selection configuration.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    keys:
+      peer_dynamic_source:
+        type: str
+        valid_values:
+          - stun
+        description: Source of dynamic peer discover. Only 'stun' is supported.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
        description: Source of dynamic peer discovery.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+      path_groups:
+        type: list
+        primary_key: name
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              description: Path group name
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
              description: Path group name.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+            id:
+              type: int
+              description: Path group ID
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
              description: Path group ID.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+              convert_types:
+                - str
+              min: 1
+              max: 65535
+            ipsec_profile:
+              type: str
+              description: IPSec profile for the path group
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
              description: IPSec profile for the path group.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+            flow_assignment:
+              type: str
+              valid_values:
+                - lan
+              description: |
+                - Flow assignement LAN can not be configured in a path group with dynamic peer
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
              description: |-
                Flow assignment `lan` can not be configured in a path group with dynamic peers.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+              primary_key: name
+              items:
+                type: dict
+                keys:
+                  name:
+                    type: str
+                    description: |
+                      Local interface name
+
+                      Ethernet<X> is supported
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]: No need to repeat the pattern.
```suggestion
                      Local interface name.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+                        min_length: 1
+                        max_length: 2
+                        description: |
+                          STUN server-profile names
+                        items:
+                          type: str
+            dynamic_peers:
+              type: dict
+              description: |
+                - Flow assignement LAN can not be configured in a path group with dynamic peer
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
              description: |-
                Flow assignment `lan` can not be configured in a path group with dynamic peers.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+              keys:
+                enabled:
+                  type: bool
+                  description: Enable `peer dynamic`
+                ip_local:
+                  type: bool
+                  description: Prefer local IP address
+                ipsec:
+                  type: bool
+                  description: IPsec configuration for dynamic peer
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
                  description: IPsec configuration for dynamic peers.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+                type: str
+      policies:
+        type: list
+        primary_key: name
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              description: Dps policy name
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
              description: DPS policy name.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+        primary_key: name
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              description: VRF name
+            path_selection_policy:
+              type: str
+              description: Dps policy name to use for this VRF
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
              description: DPS policy name to use for this VRF.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+                  stun:
+                    type: dict
+                    keys:
+                      server_profiles:
+                        type: list
+                        required: true
+                        min_length: 1
+                        max_length: 2
+                        description: |
+                          STUN server-profile names
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
                        description: STUN server-profile names
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
                        description: STUN server-profile names.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+              primary_key: ip_address
+              items:
+                type: dict
+                keys:
+                  ip_address:
+                    type: str
+                  public_address:
+                    type: str
+                    description: Public IP assigned by NAT
+                  stun:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml:77]: Add stun to the key_to_display_name.py

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 18]
```diff
@@ -0,0 +1,173 @@
...
+keys:
+  router_path_selection:
+    type: dict
+    description: Dynamic path selection configuration.
+    keys:
+      peer_dynamic_source:
+        type: str
+        valid_values:
+          - stun
+        description: Source of dynamic peer discover.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
        description: Source of dynamic peer discovery.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 19]
```diff
@@ -0,0 +1,173 @@
...
+              min: 1
+              max: 65535
+            ipsec_profile:
+              type: str
+              description: IPSec profile for the path group.
+            flow_assignment:
+              type: str
+              valid_values:
+                - lan
+              description: Flow assignement `lan` can not be configured in a path group with dynamic peer.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
              description: Flow assignement `lan` can not be configured in a path group with dynamic peers.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+        primary_key: name
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              description: Load-balance policy name.
+            path_groups:
+              type: list
+              description: List of path-group to use for this load balance policy.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]:
```suggestion
              description: List of path-groups to use for this load balance policy.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2 [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+   !
+   policy {{ policy.name }}
+{%         if policy.default_match is arista.avd.defined %}
+      default-match
+{%             if policy.default_match.load_balance is arista.avd.defined %}
+         load-balance {{ policy.default_match.load_balance }}
+{%             endif %}
+{%         endif %}
+{%         for rule in policy.rules | arista.avd.natural_sort('id') %}
+      {{ rule.id }} application-profile {{ rule.application_profile }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2:97]: No variable protection for appliation_profile

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2 [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
+{%         endif %}
+{#      local interfaces #}
+{%         for local_interface in path_group.local_interfaces | arista.avd.natural_sort('name') %}
+{%             set local_interface_def = "local interface " ~ local_interface.name %}
+{%             if local_interface.public_address is arista.avd.defined %}
+{%                 set local_interface_def = local_interface_def ~ " public address " ~ local_interface.public_address %}
+{%             endif %}
+      !
+      {{ local_interface_def }}
+{%             if local_interface.stun is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2]:
```suggestion
{%             if local_interface.stun.server_profiles is arista.avd.defined %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2 [hunk 23]
```diff
@@ -0,0 +1,109 @@
...
+{%         endfor %}
+{#      local ips #}
+{%         for local_ip in path_group.local_ips | arista.avd.natural_sort('ip_address') %}
+{%             set local_ip_def = "local ip " ~ local_ip.ip_address %}
+{%             if local_ip.public_address is arista.avd.defined %}
+{%                 set local_ip_def = local_ip_def ~ " public address " ~ local_ip.public_address %}
+{%             endif %}
+      !
+      {{ local_ip_def }}
+{%             if local_ip.stun is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2]:
```suggestion
{%             if local_ip.stun.server_profiles is arista.avd.defined %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2 [hunk 24]
```diff
@@ -0,0 +1,109 @@
...
+{%             if path_group.dynamic_peers.ipsec is arista.avd.defined(true) %}
+         ipsec
+{%             elif path_group.dynamic_peers.ipsec is arista.avd.defined(false) %}
+         ipsec disabled
+{%             endif %}
+{%         endif %}
+{#      peer static #}
+{%         for static_peer in path_group.static_peers | arista.avd.natural_sort('router_ip') %}
+      !
+      peer static {{ static_peer.router_ip }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2]: I don't feel like router_ip is the right key here. The `peer static <ip>` is some IP on the peer, and we use `static_peer.ip_address` as next hop. Maybe EOS descriptions can help us with some wording.

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2]:
```
leaf1(config-path-group-toto-0)#peer static ?
  router-ip  Peer router-IP
  ```

this actually comes fom the help message

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2]: but actuallyu it should render `peer static router-ip <router_ip?` ... :)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2 (hunk 22 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2]: [01342c9](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/01342c97cdedf6387dfbd4e71854d7734c77df4f)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml (hunk 20 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]: [01342c9](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/01342c97cdedf6387dfbd4e71854d7734c77df4f)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2 (hunk 24 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2]: [01342c9](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/01342c97cdedf6387dfbd4e71854d7734c77df4f)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2 (hunk 21 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-path-selection.j2:97]: it is required in the schema but follwoing our chat added the guard
[cf63e95](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/cf63e95e532200560ea960e434639a5d68939c11)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 25]
```diff
@@ -0,0 +1,173 @@
...
+            local_interfaces:
+              type: list
+              primary_key: name
+              items:
+                type: dict
+                keys:
+                  name:
+                    type: str
+                    description: Local interface name.
+                    pattern: "^Ethernet\\d+$"
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]: Can this not be an SVI or port-channel?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml [hunk 26]
```diff
@@ -0,0 +1,173 @@
...
+              type: list
+              primary_key: id
+              items:
+                type: dict
+                keys:
+                  id:
+                    type: int
+                    description: Rule ID.
+                    min: 1
+                    max: 255
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml:159]:
```suggestion
                    max: 255
                    convert_types:
                      - str
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2 [hunk 27]
```diff
@@ -0,0 +1,120 @@
...
+{# doc - router path-selection #}
+{% if router_path_selection is arista.avd.defined %}
+
+### Router Path-selection
+
+#### Router Path-selection Summary
+
+| Setting | Value |
+| ------  | ----- |
+{%     if router_path_selection.peer_dynamic_source is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2]:
```suggestion
{%     if router_path_selection.peer_dynamic_source is arista.avd.defined %}

#### Router Path-selection Summary

| Setting | Value |
| ------  | ----- |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2 [hunk 28]
```diff
@@ -0,0 +1,120 @@
...
+{%                 endif %}
+{%             endfor %}
+{%         endfor %}
+{%     endif %}
+{%     if router_path_selection.vrfs is arista.avd.defined %}
+
+#### VRFs configuration
+
+| VRF  name | DPS policy |
+| --------- | ---------- |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2]:
```suggestion
| VRF name | DPS policy |
| -------- | ---------- |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2 [hunk 29]
```diff
@@ -0,0 +1,120 @@
...
+{%         for load_balance_policy in router_path_selection.load_balance_policies | arista.avd.natural_sort('name') %}
+| {{ load_balance_policy.name }} | {{ load_balance_policy.path_groups | arista.avd.natural_sort() | join("<br>") }} |
+{%         endfor %}
+{%     endif %}
+{%     if router_path_selection.policies is arista.avd.defined %}
+
+#### DPS policies
+{%         for policy in router_path_selection.policies | arista.avd.natural_sort('name') %}
+
+##### DPS policy: {{ policy.name }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2]:
```suggestion
##### DPS policy {{ policy.name }}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml (hunk 25 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml]: No

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2 (hunk 27 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2]: will do it for now as there is only one setting but this won't work in the future if there are more

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2 (hunk 29 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2]: [5904680](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/59046801f25e072cff6a5a5fa1c76f19c2429c7c)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2 (hunk 28 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2]: [5904680](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/59046801f25e072cff6a5a5fa1c76f19c2429c7c)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2 (hunk 27 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2]: [5904680](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/59046801f25e072cff6a5a5fa1c76f19c2429c7c)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml (hunk 26 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_path_selection.schema.yml:159]: [5904680](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/59046801f25e072cff6a5a5fa1c76f19c2429c7c)

**Review by ClausHolbechArista — DISMISSED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-path-selection.cfg [hunk 30]
```diff
@@ -0,0 +1,84 @@
...
+      local interface Ethernet3
+         stun server-profile STUN-P-1
+      !
+      local ip 192.168.1.100 public address 192.168.42.42
+         stun server-profile STUN-P-1 STUN-P-2
+      !
+      local ip 192.168.100.1
+         stun server-profile STUN-P-1
+      !
+      peer dynamic
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-path-selection.cfg:50]: When testing this configuration on a DUT I get the following error:

```shell
eos-1(config-path-group-PG-2-42)#peer dynamic
% Flow assignement LAN can not be configured in a path group with dynamic peer.
eos-1(config-path-group-PG-2-42)#         ipsec disabled
% Invalid input
eos-1(config-path-group-PG-2-42)
```

**Comment by gmuloc** [comment]
Turns out a static peer can have multiple ip addresses - need to adjust the model

```
ceos2(config-peer-router-ip-42.42.42.42-internet)#show active
router path-selection
   path-group internet id 101
      peer static router-ip 42.42.42.42
         ipv4 address 2.2.2.2
         ipv4 address 3.3.3.3
         ipv4 address 1.1.1.1

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2 [hunk 31]
```diff
@@ -0,0 +1,121 @@
...
+| IPSec | {{ path_group.dynamic_peers.ipsec | arista.avd.default("-") }} |
+{%             endif %}
+{%             if path_group.static_peers is arista.avd.defined %}
+
+###### Static peers
+
+| Router IP | Name | IPv4 address(es) |
+| --------- | ---- | ---------------- |
+{%                 for static_peer in path_group.static_peers | arista.avd.natural_sort('router_ip') %}
+{%                     set ipv4_addresses = static_peer.ipv4_addresses | arista.avd.default([]) | join("<br>") %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2]:
```suggestion
{%                     set ipv4_addresses = static_peer.ipv4_addresses | arista.avd.default(["-"]) | join("<br>") %}
```

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-path-selection.j2]: [d21c504](https://github.com/aristanetworks/ansible-avd/pull/3203/commits/d21c50487c16d4528558d8b67f066a987594af30)

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**